File: content/LLD/Reference-Code/3. Memento Pattern/text_editor.py
from text_memento import TextMemento


class TextEditor:
    def __init__(self):
        self.__text = ""

    def write(self, new_text):
        self.__text += new_text

    def get_text(self) -> str:
        return self.__text

    def save(self) -> TextMemento:
        return TextMemento(self.__text)

    def restore(self, tm: TextMemento):
        self.__text = tm.get_saved_text()

File: content/LLD/Reference-Code/3. Memento Pattern/TextEditor.java
public class TextEditor {
    private String text = "";

    public void write(String newText) {
        this.text += newText;
    }

    public String getText() {
        return this.text;
    }

    public TextMemento save() {
        return new TextMemento(this.text);
    }

    public void restore(TextMemento tm) {
        this.text = (tm != null) ? tm.getSavedText() : "";
    }
}
